HousingRegisterApi/V1/UseCase/RecalculateBedroomsUseCase.cs

```diff
@@ -0,0 +1,81 @@
+using HousingRegisterApi.V1.Boundary.Request;
+using HousingRegisterApi.V1.Gateways;
+using HousingRegisterApi.V1.UseCase.Interfaces;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+
+namespace HousingRegisterApi.V1.UseCase
+{
+    public class RecalculateBedroomsUseCase : IRecalculateBedroomsUseCase
+    {
+        private readonly ILogger _logger;
+        private readonly IApplicationApiGateway _gateway;
+        public RecalculateBedroomsUseCase(ILogger<RecalculateBedroomsUseCase> logger, IApplicationApiGateway gateway)
+        {
+            _logger = logger;
+            _gateway = gateway;
+        }
+
+        public void Execute()
+        {
+            var searchParameters = new SearchQueryParameter()
+            {
+                Status = "Submitted",
+                HasAssessment = true,
+            };
+
+            _logger.LogInformation($"Searching for submitted applications with assessments");
+            var applications = _gateway.GetApplications(searchParameters);
+
+            if (!applications.Any())
+            {
+                _logger.LogInformation($"No applications found");
+                return;
+            }
+
+            // code comes here if applications are found
+            _logger.LogInformation($"Recalculating bedrooms needed for {applications.Count()} applications");
+
+            foreach (var application in applications)
+            {
+                try
+                {
+                    if (application.Assessment == null)
+                    {
+                        throw new InvalidOperationException($"No assessment exists");
+                    }
+
+                    var currentBedroomNeed = application.Assessment.BedroomNeed;
+                    application.RecalulateBedrooms();
+                    var newBedroomNeed = application.Assessment.BedroomNeed;
+
+                    // only update if required
+                    if (newBedroomNeed != currentBedroomNeed)
+                    {
+                        application.Assessment.BedroomNeed = newBedroomNeed;
+                        UpdateApplicationRequest updateRequest = new UpdateApplicationRequest()
+                        {
+                            Assessment = application.Assessment,
```

Comment:
For an `UpdateApplicationRequest` we should only need to provide the changed data, as the gateway should ignore updating other values if they are empty.